app/src/main/java/org/wikipedia/analytics/eventplatform/EditAttemptStepEvent.kt

```diff
@@ -0,0 +1,74 @@
+package org.wikipedia.analytics.eventplatform
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.analytics.eventplatform.EditorInterfaceType.OTHER
+
+@Serializable
+@SerialName("/analytics/legacy/editattemptstep/1.2.0")
+class EditAttemptStepEvent(private val event: EditAttemptStepInteractionEvent) : Event(STREAM_NAME) {
+
+    companion object {
+
+        fun logInit() {
+            submitEditAttemptEvent(ActionType.INIT)
+        }
+
+        fun logSaveIntent() {
+            submitEditAttemptEvent(ActionType.SAVE_INTENT)
+        }
+
+        fun logSaveAttempt() {
+            submitEditAttemptEvent(ActionType.SAVE_ATTEMPT)
+        }
+
+        fun logSaveSuccess() {
+            submitEditAttemptEvent(ActionType.SAVE_SUCCESS)
+        }
+
+        fun logSaveFailure() {
+            submitEditAttemptEvent(ActionType.SAVE_FAILURE)
+        }
+
+        private fun submitEditAttemptEvent(action: ActionType) {
+            EventPlatformClient.submit(EditAttemptStepEvent(EditAttemptStepInteractionEvent(action.valueString, "example",
+                OTHER.valueString, INTEGRATION_ID, "example", PLATFORM, 0, 1, 1)))
+        }
+
+        private const val STREAM_NAME = "eventlogging_EditAttemptStep"
+        private const val INTEGRATION_ID = "app-android"
+        private val PLATFORM = WikipediaApp.getInstance().getString(R.string.device_type).lowercase()
```

Comment:
Would it be possible to have these variables at the top of the `companion object`.